scrappers/itaka/scraper.py

```diff
@@ -0,0 +1,202 @@
+import time
+import requests
+import json
+import random
+
+from bs4 import BeautifulSoup
+from typing import List, Tuple
+
+from selenium.webdriver.common.action_chains import ActionChains
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+
+WEBSITE_URL: str = "https://www.itaka.pl/all-inclusive/"
+NO_OF_SCRAPED_OFFERS: int = 500
+
+
+class ItakaScraper:
+    def __init__(self, website_url: str, no_of_scraped_offers: int) -> None:
+        self.website_url: str = website_url
+        self.no_of_scraped_offers: int = no_of_scraped_offers
+        self.offer_urls: List[str] = self.get_itaka_urls()
+        self.scraped_data: List[dict] = self.generate_data()
+        self.save_json_as_file(filename="itaka_offers")
+
+    def _initialize_driver(self) -> webdriver.Chrome:
+        driver = webdriver.Chrome()
+        driver.get(self.website_url)
+        return driver
+
+    def _accept_cookies(self, driver: webdriver.Chrome) -> None:
+        time.sleep(5)
+        accept_agreement_element = WebDriverWait(driver, 5).until(
+            EC.presence_of_element_located(
+                (
+                    By.XPATH,
+                    "/html/body/div[5]/div[2]/div[2]/div/div[3]/button[2]",
+                )
+            )
+        )
+        accept_agreement_element.click()
+        time.sleep(5)
+        accept_cookies_element = driver.find_element(
+            By.XPATH, '//*[@id="cookie-info-container"]/div/p/button'
+        )
+        accept_cookies_element.click()
+        time.sleep(5)
+
+    def _website_scroll(self, driver: webdriver.Chrome) -> None:
+        generate_more_content_iterations = self.no_of_scraped_offers // 25
+        for _ in range(generate_more_content_iterations - 1):
+            generate_more_offers_button = driver.find_element(
+                By.XPATH,
+                '//*[@id="search-results-container"]/div/div/section/div/div[1]/div[2]/div[1]',
+            )
+            actions = ActionChains(driver)
+            actions.move_to_element(generate_more_offers_button).perform()
+            time.sleep(3)
+            generate_more_offers_button.click()
+            time.sleep(3)
+
+    def _generate_url_list(self, driver: webdriver.Chrome) -> List[str]:
+        html_page_source = driver.page_source
+        html_content = BeautifulSoup(html_page_source, "html.parser")
+        offer_titles = html_content.find_all("h3", {"class": "header_title"})
+        offer_urls = [
+            f"https://www.itaka.pl{title.find('a')['href']}" for title in offer_titles
+        ]
+
+        return offer_urls
```

Comment:
niepotrzeba zmienna `offer_urls`
```suggestion
        return [
            f"https://www.itaka.pl{title.find('a')['href']}" 
            for title in offer_titles
        ]

```